### DISCUSSION TITLE
How to draw image relatively?

### DISCUSSION THREAD

**Discussion by WgZgYn** [hajimehoshi/ebiten#2825] [Q&A] [answered]
For example, if want draw elements in the block or the layout like the grid, what should I do? I know the ebiten.DrawImageOptions struct, but I don't know how to set the original point, every time I call op.GeoM.Reset() the point comes to (0, 0). I hope there a Postion struct construct with const original point, and do the same thing as DrawImageOptions, but the Reset() could recover the original point not (0, 0). In my game I design a simple Pos struct, I want to know whether ebitengine have a similar struct? Whether it is necessary?

**Comment by hajimehoshi** [comment]
What about

```go
op.GeoM.Reset()
op.GeoM.Translate(originalX, originalY)
```

?

**Reply by WgZgYn** [reply]
yeap thank you. I konw the `op.GeoM.Translate(originalX, originalY)`. 
But firstly the op most time is a pointer value, if I pass the `op` to a function, it is maybe changed, and I don't know a good way to make a copy of the op. 
And, I don't know how to get the current point from the DrawImageOption directly to save the originalX and originalY, when it comes to complex layout calculating absolute postion is much difficult. 
I am confused about the game programming style.   ( x _ x )

**Reply by hajimehoshi** [reply, marked as answer]
You can copy GeoM value and use this.

```go
var g0 GeoM
g0.Translate(1, 2)
g1 := g0 // g0 is copied.
g0.Rotate(1) // this doesn't affect g1
```

**Reply by WgZgYn** [reply]
I will try it. Thanks😃
